lua/entities/gmod_tardis/modules/sh_health.lua

```diff
@@ -15,33 +15,54 @@ ENT:AddHook("SettingChanged","maxhealth-changed", function(self, id, val)
     end
 end)
 
-function ENT:ChangeHealth(newhealth)
+function ENT:ChangeHealth(new_health)
+
+    if CLIENT then
+        self:SendMessage("health_change", {new_health})
+        return
+    end
+
     if self:GetRepairing() then
         return
     end
-    local maxhealth = TARDIS:GetSetting("health-max")
+    local max_health = TARDIS:GetSetting("health-max")
     if not TARDIS:GetSetting("health-enabled") then
-        self:SetData("health-val", maxhealth, true)
+        self:SetData("health-val", max_health, true)
         return
     end
-    local oldhealth = self:GetHealth()
-    if newhealth > oldhealth and oldhealth+newhealth > maxhealth then
-        newhealth = maxhealth
-    end
-    if newhealth <= 0 then
-        newhealth = 0
-        if newhealth ~= oldhealth then
-            self:CallCommonHook("OnHealthDepleted")
-        end
+    local old_health = self:GetHealth()
+
+    new_health = math.Clamp(new_health, 0, max_health)
+
+    self:SetData("health-val", new_health, true)
+    self:CallCommonHook("OnHealthChange", new_health, old_health)
+    if new_health == 0 and old_health ~= 0 then
+        self:CallCommonHook("OnHealthDepleted")
     end
-    self:SetData("health-val", newhealth, true)
-    self:CallCommonHook("OnHealthChange", newhealth, oldhealth)
+end
+
+if SERVER then
+    ENT:OnMessage("health_change", function(self, data, ply)
+        self:ChangeHealth(data[1])
+    end)
 end
 
 function ENT:GetHealth()
     return self:GetData("health-val", 0)
 end
 
+function ENT:IsAlive()
+    return (self:GetHealth() ~= 0)
+end
+
+function ENT:GetRepairPrimed()
```

Comment:
This code is duplicated from fixes already made in dev for both `GetRepairPrimed` and `GetRepairing`

Reply:
Rebase conflict, my bad

Reply:
9115edec229fd2a660e9f7d8946ec6f94d1ef27f